test/SlowTests/Client/Attachments/AttachmentsReplication.cs

```diff
@@ -2529,144 +2527,138 @@ public async Task ConflictOfAttachmentAndDocument3StoresDifferentLastModifiedOrd
-                    var u = await session.LoadAsync<User>("users/1");
-                    u.Age = 30;
-                    await session.SaveChangesAsync();
-                }
+            await EnsureReplicatingAsync(store1, store2);
+            await EnsureReplicatingAsync(store1, store3);
 
-                await EnsureReplicatingAsync(store1, store2);
-                await EnsureReplicatingAsync(store1, store3);
+            using (var backgroundStream = new MemoryStream(new byte[] { 10, 20, 30, 40, 50 }))
+            {
+                var result = store2.Operations.Send(new PutAttachmentOperation("users/1", "foo/bar", backgroundStream, "image/png"));
+                Assert.Equal("foo/bar", result.Name);
+                Assert.Equal("users/1", result.DocumentId);
+                Assert.Equal("image/png", result.ContentType);
+                Assert.Equal("igkD5aEdkdAsAB/VpYm1uFlfZIP9M2LSUsD6f6RVW9U=", result.Hash);
+            }
 
-                await WriteStatus(stores, "Stage 2");
+            using (var backgroundStream = new MemoryStream(new byte[] { 10, 20, 30, 40, 50, 60 }))
+            {
+                var result = store3.Operations.Send(new PutAttachmentOperation("users/1", "foo/bar", backgroundStream, "image/png"));
+                Assert.Equal("foo/bar", result.Name);
+                Assert.Equal("users/1", result.DocumentId);
+                Assert.Equal("image/png", result.ContentType);
+                Assert.Equal("7hoAZadly0e2TKk4NC6+MrtVuqZblV3+UDW7/Iz9H5U=", result.Hash);
+            }
 
-                await SetupReplicationAsync(store2, store1);
-                await SetupReplicationAsync(store2, store3);
-                await EnsureReplicatingAsync(store2, store1);
-                await EnsureReplicatingAsync(store2, store3);
+            using (var session = store1.OpenAsyncSession())
+            {
+                var u = await session.LoadAsync<User>("users/1");
+                u.Age = 30;
+                await session.SaveChangesAsync();
+            }
 
-                await WriteStatus(stores, "Stage 3");
+            await EnsureReplicatingAsync(store1, store2);
+            await EnsureReplicatingAsync(store1, store3);
 
-                await SetupReplicationAsync(store3, store2);
-                await EnsureReplicatingAsync(store3, store2);
+            await SetupReplicationAsync(store2, store1);
+            await SetupReplicationAsync(store2, store3);
 
-                await WriteStatus(stores, "Stage 4");
+            await EnsureReplicatingAsync(store2, store1);
+            await EnsureReplicatingAsync(store2, store3);
 
-                var res2 = await WaitForChangeVectorInReplicationAsync(stores);
-                Assert.True(res2);
+            await SetupReplicationAsync(store3, store2);
+            await EnsureReplicatingAsync(store3, store2);
 
-                var res = await WaitForValueAsync(async () =>
+            var res = await WaitForValueAsync(async () =>
```

Comment:
value of res is not checked anywhere

Reply:
we are waiting here for the results to be consistent, but we check the actual details in the following lines